python/paddle/fluid/tests/unittests/test_matmul_v2_op.py

```diff
@@ -401,148 +350,152 @@ def test_dygraph_fp16(self):
             place = core.CUDAPlace(0)
             if core.is_float16_supported(place):
                 with fluid.dygraph.guard(place):
-                    input_x = np.random.random([4, 3]).astype("float16")
-                    input_y = np.random.random([3, 4]).astype("float16")
+                    input_x = np.random.random([4, 3]).astype('float16')
+                    input_y = np.random.random([3, 4]).astype('float16')
                     x = paddle.to_tensor(input_x)
                     y = paddle.to_tensor(input_y)
                     result = paddle.matmul(x, y)
 
+    def test_api_eager_dygraph(self):
+        with _test_eager_guard():
+            self.test_dygraph()
+            self.test_dygraph_fp16()
+
 
 class TestComplexMatMulOp(OpTest):
     def setUp(self):
-        self.op_type = "matmul_v2"
+        self.op_type = 'matmul_v2'
         self.init_base_dtype()
         self.init_input_output()
         self.init_grad_input_output()
-
         self.inputs = {
             'X': OpTest.np_dtype_to_fluid_dtype(self.x),
             'Y': OpTest.np_dtype_to_fluid_dtype(self.y)
         }
-        self.attrs = {'axis': -1, 'use_mkldnn': False}
+        self.attrs = {'axis': (-1), 'use_mkldnn': False}
         self.outputs = {'Out': self.out}
 
     def init_base_dtype(self):
         self.dtype = np.float64
 
     def init_input_output(self):
-        self.x = np.random.random(
-            (10, 10)).astype(self.dtype) + 1J * np.random.random(
-                (10, 10)).astype(self.dtype)
-        self.y = np.random.random(
-            (10, 10)).astype(self.dtype) + 1J * np.random.random(
-                (10, 10)).astype(self.dtype)
+        self.x = (np.random.random((10, 10)).astype(self.dtype) +
+                  (1j * np.random.random((10, 10)).astype(self.dtype)))
+        self.y = (np.random.random((10, 10)).astype(self.dtype) +
+                  (1j * np.random.random((10, 10)).astype(self.dtype)))
         self.out = np.dot(self.x, self.y)
 
     def init_grad_input_output(self):
-        self.grad_out = np.ones((10, 10), self.dtype) + 1J * np.ones(
-            (10, 10), self.dtype)
+        self.grad_out = (np.ones((10, 10), self.dtype) + (1j * np.ones(
+            (10, 10), self.dtype)))
         self.grad_x = np.matmul(self.grad_out, np.conj(self.y).T)
         self.grad_y = np.matmul(np.conj(self.x).T, self.grad_out)
 
     def test_check_output(self):
-        self.check_output()
+        self.check_output(check_eager=True)
 
     def test_check_grad_normal(self):
         self.check_grad(
             ['X', 'Y'],
             'Out',
             user_defined_grads=[self.grad_x, self.grad_y],
-            user_defined_grad_outputs=[self.grad_out])
+            user_defined_grad_outputs=[self.grad_out],
+            check_eager=False)
```

Comment:
why check_eager is false here？

Reply:
We haven't supported Type Promotion at Dygraph level yet, so I turned of eager mode test for "test_check_grad_normal" and "test_check_grad_ignore_y" for now.